### PR TITLE
misc: Use external_id instead of customer_id and subscription_id

### PR DESCRIPTION
**Pull request by rsempe**
## Context

We want to clarify the difference between lago ids and external ids.

Today, we have `customer_id` on customers and `subscription_id|unique_id` on subscriptions.

On the API, the goal is to use external_ids:
- `external_id` or `external_customer_id` for customer
- `external_id` or `external_subscription_id` for subscription

## Description

The goal of this PR is to:
- rename `customer_id` to `external_id` on customers
- rename `subscription_id` to `external_id` on subscriptions
- remove `unique_id` from subscriptions

### CODE REVIEW

File: app/services/subscriptions/create_service.rb [hunk 1]
```diff
@@ -40,10 +40,9 @@ def create(**args)
 
       # NOTE: prepare subscription attributes
       @name = args[:name]&.strip
-      @unique_id = SecureRandom.uuid
+      @external_id = SecureRandom.uuid
```
Comment (vincent-pochet) [inline app/services/subscriptions/create_service.rb:43]: Wondering if this still apply. Should we put the customer external id (Or waiting from the front to provide this value)?

File: app/services/invoices/customer_usage_service.rb [hunk 2]
```diff
@@ -2,20 +2,19 @@
 
 module Invoices
   class CustomerUsageService < BaseService
+    # NOTE: customer_id can reference the lago id or the external id of the customer
+    # NOTE: subscription_id can reference the lago id or the external id of the subscription
```
Comment (vincent-pochet) [inline app/services/invoices/customer_usage_service.rb:6]: Merf :/
Something that we might improve in a near future :smile:

File: app/services/payment_provider_customers/create_service.rb [hunk 3]
```diff
@@ -10,7 +10,7 @@ def initialize(customer)
 
     def create_or_update(customer_class:, payment_provider_id:, params:, async: true)
       provider_customer = customer_class.find_or_initialize_by(
-        customer_id: customer.id,
+        customer_id: customer.id, # TODO: Why?
```
Comment (vincent-pochet) [inline app/services/payment_provider_customers/create_service.rb:13]: :thinking:

**Review by vincent-pochet — APPROVED** [review]
:exploding_head:

File: app/services/payment_provider_customers/create_service.rb (hunk 3 above)
Comment (vincent-pochet) [reply app/services/payment_provider_customers/create_service.rb:13]: `customer_class` will be a `PaymentProviderCustomers::StripeCustomer` so no changes required here. (Appart from a refact maybe... :D )

File: db/schema.rb [hunk 4]
```diff
@@ -327,7 +327,7 @@
     t.uuid "previous_subscription_id"
     t.date "subscription_date"
     t.string "name"
-    t.string "unique_id", null: false
+    t.string "external_id", null: false
```
[the remark this answers is not in the document]
Comment (rsempe) [reply db/schema.rb:330]: You're right - Added in that PR even if not related 👍

File: app/services/subscriptions/create_service.rb (hunk 1 above)
Comment (rsempe) [reply app/services/subscriptions/create_service.rb:43]: Well - Not sure we want the front to manage external ids and not sure we want to store the external customer id.

For me, I think it's good enough to generate this info from the backend, no?

File: app/services/invoices/customer_usage_service.rb (hunk 2 above)
Comment (rsempe) [reply app/services/invoices/customer_usage_service.rb:6]: Yes - That's why I put a note here 😅

File: app/services/payment_provider_customers/create_service.rb (hunk 3 above)
Comment (rsempe) [reply app/services/payment_provider_customers/create_service.rb]: TODO removed 😅

**Pull request merged**